.ci/cico_build_deploy_test_rhche.sh

```diff
@@ -0,0 +1,166 @@
+}
+
+# Retrieve and test credentials
+set +x
+eval "$(./env-toolkit load -f jenkins-env.json -r \
+        ^DEVSHIFT_TAG_LEN$ \
+        ^QUAY_ \
+        ^KEYCLOAK \
+        ^BUILD_NUMBER$ \
+        ^JOB_NAME$ \
+        ^ghprbPullId$ \
+        ^RH_CHE)"
+
+if [[ "${PR_CHECK_BUILD}" == "true" ]]; then
+	echo "Running ${JOB_NAME} PR: #${ghprbPullId}, build number #${BUILD_NUMBER}, testing creds:"
+else
+	echo "Running ${JOB_NAME} with build, deploy and test."
+fi
+
+
+CREDS_NOT_SET="false"
+curl -s "https://mirror.openshift.com/pub/openshift-v3/clients/${OC_VERSION}/linux/oc.tar.gz" | tar xvz -C /usr/local/bin
+
+if [[ -z "${QUAY_USERNAME}" && -z "${QUAY_PASSWORD}" ]]; then
+  echo "Docker registry credentials not set"
+  CREDS_NOT_SET="true"
+fi
+
+if [[ -z "${RH_CHE_AUTOMATION_DEV_CLUSTER_SA_TOKEN}" ]]; then
+  echo "RDU2C credentials not set"
+  CREDS_NOT_SET="true"
+fi
+
+if [[ -z "${KEYCLOAK_TOKEN}" || -z "${RH_CHE_AUTOMATION_CHE_PREVIEW_USERNAME}" || -z "${RH_CHE_AUTOMATION_CHE_PREVIEW_PASSWORD}" ]]; then
+  echo "Prod-preview credentials not set."
+  CREDS_NOT_SET="true"
+fi
+
+if [ "${CREDS_NOT_SET}" = "true" ]; then
+  echo "Failed to parse jenkins secure store credentials"
+  exit 2
+else
+  echo "Credentials set successfully."
+fi
+
+if oc login ${DEV_CLUSTER_URL} --insecure-skip-tls-verify \
+                               --token "${RH_CHE_AUTOMATION_DEV_CLUSTER_SA_TOKEN}";
+then
+  echo "OpenShift login successful"
+else
+  echo "OpenShift login failed"
+  exit 3
+fi
+
+set -x
+
+# Getting core repos ready
+yum install --assumeyes epel-release
+yum update --assumeyes
+yum install --assumeyes python-pip
+
+# Test and show version
+pip -V
+
+# Getting dependencies ready
+yum install --assumeyes \
+            docker \
+            git \
+            patch \
+            pcp \
+            bzip2 \
+            golang \
+            make \
+            jq \
+            java-1.8.0-openjdk \
+            java-1.8.0-openjdk-devel \
+            centos-release-scl
+
+yum install --assumeyes \
+            rh-maven33 \
+            rh-nodejs4
+
+systemctl start docker
+pip install yq
+
+set +x
+#change version of used che
+if [[ "${USE_CHE_LATEST_SNAPSHOT}" == "true" ]]; then 
+	CHE_VERSION=$(curl -s https://raw.githubusercontent.com/eclipse/che/master/pom.xml | grep "^    <version>.*</version>$" | awk -F'[><]' '{print $3}')
+	echo ">>> change upstream version to: $CHE_VERSION"
+	mvn versions:update-parent  versions:commit -DallowSnapshots=true -DparentVersion=[$CHE_VERSION] -U
+	namespace=compatibility-check
+	dockerImageTag=$DOCKER_IMAGE_TAG
+else
+	namespace=prcheck-${RH_PULL_REQUEST_ID}
+	dockerImageTag="${RH_TAG_DIST_SUFFIX}"-"${RH_PULL_REQUEST_ID}"
+fi
+
+# Build and push image to docker registry
+source ./config
+BuildTagAndPushDocker
+
+# Deploy rh-che image
+echo "Deploying nightly-${RH_TAG_DIST_SUFFIX} from ${DOCKER_IMAGE_URL}"
+
+if ./dev-scripts/deploy_custom_rh-che.sh -o "${RH_CHE_AUTOMATION_DEV_CLUSTER_SA_TOKEN}" \
```

Comment:
As far as I understand in this case all the output of that script is suppressed. Shouldn't we let script log all its logs to be able to trace it in a job output in a case something breaks?

Reply:
Sorry, I'm not sure what you mean about that. The messages that are printed are readable in log, e.g.:
```
Deploying upstream-check-latest from quay.io/openshiftio/rhel-rhchestage-rh-che-automation
[1mLogging in to openshift...[0m
Login successful, creating project compatibility-check
Switched to project compatibility-check
[92;1mProject created successfully.[0;38;5;238m
[0;1mGetting deployment scripts...[0m
[92;1mGetting deployment scripts done.[0m
Waiting for postgres deployment to be deleted...................
Postgres deployment successfully deleted.
[1mDeploying PostgreSQL[0;38;5;60m
[0;38;5;60mPostgres status: Available:[0;1m"False" [0;38;5;60mProgressing:[0;1m"Unknown" [0;38;5;60mTimeout:[0;1m180[0m
[0;38;5;60mPostgres status: Available:[0;1m"False" [0;38;5;60mProgressing:[0;1m"Unknown" [0;38;5;60mTimeout:[0;1m179[0m
[0;38;5;60mPostgres status: Available:[0;1m"False" [0;38;5;60mProgressing:[0;1m"Unknown" [0;38;5;60mTimeout:[0;1m178[0m
...
```
You mean that this is not enough and there should be any other logs?